PWGHF/Tasks/taskD0parametrizedPID.cxx

```diff
@@ -113,13 +118,51 @@ struct TaskD0parametrizedPIDMC {
       //if (ncontributor<=centralitySelectionMin && ncontributor>centralitySelectionMax) {
       //  continue;
       //}
+      Double_t maxFiducialY = 100.0;
+      Double_t minFiducialY = -100.0;
       if (std::abs(particle.flagMCMatchGen()) == 1 << DecayType::D0ToPiK) {
         if (std::abs(RecoDecay::Y(array{particle.px(), particle.py(), particle.pz()}, RecoDecay::getMassPDG(particle.pdgCode()))) > 4.0) {
           continue;
         }
         auto ptGen = particle.pt();
         auto yGen = RecoDecay::Y(array{particle.px(), particle.py(), particle.pz()}, RecoDecay::getMassPDG(particle.pdgCode()));
         registry.fill(HIST("hMassGen"), ptGen, std::abs(yGen));
+        if (ptGen < 5.0) {
+          maxFiducialY = -0.2 / 15 * ptGen * ptGen + 1.9 / 15 * ptGen + 0.5;
+          minFiducialY = 0.2 / 15 * ptGen * ptGen - 1.9 / 15 * ptGen - 0.5;
+        }
+        if (ptGen > 5.0) {
+          maxFiducialY = 0.8;
+          minFiducialY = -0.8;
+        }
+
+        float etaProngs[2];
+        for (int iD = particle.daughter0Id(), counter = 0; iD <= particle.daughter1Id(); ++iD, counter++) {
+          auto daught = particlesMC.iteratorAt(iD);
+          etaProngs[counter] = std::abs(daught.eta());
+        }
+
+        if (std::abs(yGen) < 2.5) {
+          registry.fill(HIST("hGenFixCMSYNoEta"), ptGen);
+        }
+
+        if (yGen > minFiducialY && yGen < maxFiducialY) {
+          registry.fill(HIST("hGenPtDepYNoEta"), ptGen);
+        }
+
+        if (etaProngs[0] < 2.5 && etaProngs[1] < 2.5) {
+          if (std::abs(yGen) < 2.5) {
+            registry.fill(HIST("hGenFixCMSYCMSEta"), ptGen);
+          }
+          if (yGen > minFiducialY && yGen < maxFiducialY) {
+            registry.fill(HIST("hGenPtDepYCMSEta"), ptGen);
+          }
+          if (etaProngs[0] < 0.8 && etaProngs[1] < 0.8) {
+            if (yGen > minFiducialY && yGen < maxFiducialY) {
+              registry.fill(HIST("hGenPtDepYALICE2Eta"), ptGen);
+            }
+          }
```

Comment:
L160-164: it's not fully clear to me why you nest this loop inside the one at L15. Of course it works equally fine, but would be more clear having it outside (otherwise, if you want to keep it nested, at this point you could just move it after L158, inside the already existing fidutial acceptance 'if', removing the repeated fiductial acceptance check).
Also, I see that don't you have the corresponding hGenFixYALICE2Eta histogram, it's by mistake or maybe just because you don't need it (i.e. in ALICE we always use the yFiducial cut)?

Reply:
I put the L160-164 inside the if (etaProngs[0] < 2.5 && etaProngs[1] < 2.5) {} to avoid the checking for tracks with |eta| > 2.5. Yes, I can move the condition inside the if condition in line 157. I do not need the hGenFixYALICE2Eta histogram for the calculation. I think for the fiducial correction factor in ALICE we generally use 2*yFiducial cut with the assumption that the rapidity distribution is flat. Correct me if I am wrong. But I added these histograms are to calculate the correction factor for going from ALICE 1 to CMS-like acceptance eta < 2.5, as suggested by Marco:
yield_(y_D < x, eta_daughter<2.5)/yield_(y_D < x, eta_daughter < 0.8) * yield_(Y_D < 2.5, eta daughter < 2.5)/yield(Y_D < 2.5, eta daughter anywhere)

@fcolamar are we missing any information for the correction?

Reply:
Hi @skundu692, I didn't have the time to process in detail Marco's correction, but if that's the one to be used, it would seem that you added everything. So for what concerns the taskD0 updates, what you have now is fine to me.